### ISSUE TITLE
error calling datasource: Couldn't read datasource - The filename, directory name, or volume label syntax is incorrect.

### ISSUE THREAD

**Issue by markus-wa** [hairyhenderson/gomplate#505]
Hi

First of all, thanks for this project, it's very neat!

Regarding my issue. In Git bash as well as CMD on Windows I get the following error:

```
$ cat config.yaml
foo:
  bar:
    baz: qux
$ gomplate -d config=./config.yaml -i 'the value we want is: {{ (datasource "config").foo.bar.baz }}'
the value we want is: template: :1:26: executing "" at : error calling datasource: Couldn't read datasource 'config': Can't stat \C:\Users\walthema\config.yaml: GetFileAttributesEx \C:\Users\walthema\config.yaml: The filename, directory nam e, or volume label syntax is incorrect.
```

~~Seeing how I'm the only person with this problem I suspect something about my environment not being right. Do you have any idea what it could be?~~

**Comment by rayjlinden** [comment]
I typed in the same thing and worked fine for me.  (Although at first I had made the file config.yml instead .yaml which did create the same error.)

**Comment by hairyhenderson** [comment]
@markus-wa First off, you're welcome! 🙂

Does this work?

```console
$ gomplate -d config.yaml -i 'the value we want is: {{ (datasource "config").foo.bar.baz }}'
```

I wonder if the `./` is interpreted in a strange way on Windows...

Unfortunately I don't have a Windows machine to develop on so mostly I rely on the limited tests that are enabled for the AppVeyor build 😉. Clearly I need to add some of the integration tests to that one!

**Comment by markus-wa** [comment]
No luck unfortunately @rayjlinden / @hairyhenderson 

```
$ gomplate -d config.yaml -i 'the value we want is: {{ (datasource "config").foo.bar.baz }}'
the value we want is: template: <arg>:1:26: executing "<arg>" at <datasource "config">: error calling datasource: Couldn't read datasource 'config': Can't stat \C:\Users\walthema\config.yaml: GetFileAttributesEx \C:\Users\walthema\config.yaml: The filename, directory name, or volume label syntax is incorrect.

# yaml -> yml
$ mv config.yaml config.yml

$ gomplate -d config.yml -i 'the value we want is: {{ (datasource "config").foo.bar.baz }}'
the value we want is: template: <arg>:1:26: executing "<arg>" at <datasource "config">: error calling datasource: Couldn't read datasource 'config': Can't stat \C:\Users\walthema\config.yml: GetFileAttributesEx \C:\Users\walthema\config.yml: The filename, directory name, or volume label syntax is incorrect.

$ gomplate -d config=./config.yml -i 'the value we want is: {{ (datasource "config").foo.bar.baz }}'
the value we want is: template: <arg>:1:26: executing "<arg>" at <datasource "config">: error calling datasource: Couldn't read datasource 'config': Can't stat \C:\Users\walthema\config.yml: GetFileAttributesEx \C:\Users\walthema\config.yml: The filename, directory name, or volume label syntax is incorrect.


```

**Comment by MikhailSemichev** [comment]
The same issue is reproduced on my Windows development machine as well.
From first glance the problem is in incorrect paths being formed by tool:
```
Couldn't read datasource 'config': Can't stat \C:\Users\walthema\config.yaml
```
It looks like the first slash in path "\C:\Users" should be removed and correct path should be:
```
C:\Users\walthema\config.yaml
```

**Comment by hairyhenderson** [comment]
Hi @MikhailSemichev - thanks for adding to the discussion! I'm working on a few fixes in #527 this weekend - it's been quite a challenge getting a Windows 10 VM up and running so I can test gomplate in it 😉 - but, I'm making progress and will try to get this bug fixed before the next release!

**Comment by markus-wa** [comment]
Thanks a lot, the fix works like a charm!
